packages/cli/src/commands/init/index.ts

```diff
@@ -47,5 +47,10 @@ export default {
       description:
         'Inits a project with a custom package name (Android) and bundle ID (iOS), e.g. com.example.app',
     },
+    {
+      name: '--platform-name <string>',
+      description:
+        'Name of out of tree platform to be used for ex. react-native-macos. This flag is optional as it should be passed automatically by out of tree platform.',
+    },
```

Comment:
Let's add this flag to docs.

Reply:
I think it's worth to be specific in the docs that `platform-name` needs to match the fork name, to avoid trying to use it with e.g. `visionos`

Reply:
+1, we should mention that it should be equal to the `name` property specified in `package.json`